SelfDriveEnv.py

```diff
@@ -270,79 +154,394 @@ def handle_events(self):
+    are not rendered and do not exist from the perspective of the car
+
+    Attributes:
+        dimenstions:    ((int, int, int, int))
+            The x, y, width and height of the TrackBorder in pixels    
+        rect:  (pygame.rect)
+            Pygame rect representing this class
+        start_color:    (int, int , int)
+            Color of starting line TrackBorders
+        finish_color:   (int, int, int)
+            Color of finish line TrackBorders
+        default_color:  (int, int, int)
+            Color of regular active TrackBorders        
+        border_color:   (int, int, int)
+            Color of the outline on a TrackBorders        
+        active: (bool)
+            Represents whether or not the TrackBorder should be used
+            rendering/collisions
+        start_finish:   (string | None)
+            Denotes whether the track is used in a starting line ("start"),
+            finish line ("finish") or neither (None)
+        index:          ([int, int])
+            A list containing the row, col position of this TrackBorder in an
+            array of TrackBorders
+
+        mutable:        (bool)
+            False to lock a TrackBorder from changes (like mouse clicks, for example)
+    """
+    
+    def __init__(self, x, y, width, height, index):
+        
+        self.dimensions = (x, y, width, height)
+        self.rect = pygame.Rect(*self.dimensions)
+        
+        self.start_color = cfg.track["start_line_color"]
+        self.finish_color = cfg.track["finish_line_color"]
+        self.default_color = cfg.track["default_color"]
+        self.color = self.default_color
+        self.border_color = cfg.track["border_color"]
+        
+        self.active = True
+        self.start_finish = None
+        self.index = index
+        self.mutable = True
+
+    def check_state(self):
+        mouse_pos = pygame.mouse.get_pos()
+        pressed_state = pygame.mouse.get_pressed()
+        keys = pygame.key.get_pressed()
+        x, y, w, h = self.dimensions
+        if x+w > mouse_pos[0] > x and y + h > mouse_pos[1] > y and self.mutable:
+            if pressed_state[0]: 
+                self.active = False
+                self.start_finish = None                
+            elif pressed_state[2]:
+                self.active = True
+                self.start_finish = None
+            elif keys[pygame.K_s]:
+                self.start_finish = "start"
+            elif keys[pygame.K_f]:
+                self.start_finish = "finish"
+        if self.start_finish == "start":
+            self.color = self.start_color
+        elif self.start_finish == "finish":
+            self.color = self.finish_color
+        else: self.color = self.default_color
+    
+    def render(self, screen):
+        self.check_state()
+        if self.active:        
+            pygame.draw.rect(screen, self.color, self.rect)    
+            pygame.draw.rect(screen, self.border_color, self.rect, 1)
```

Comment:
I was thinking for this TrackBorder class, since it is only used within the Track class as an element that makes up the Track, why not make the TrackBorder like inside the Track class: like a nested class.

Reply:
Yeah, good point. I think for now it might benefit us to keep it separate until we do the physics stuff just because the track class is so long and we're probably going to rewrite a lot of it anyway.